swig/lightgbmlib.i

```diff
@@ -133,6 +133,17 @@
           int size;
   };
 
+  int check_exception(JNIEnv *jenv) {
+    jboolean flag = jenv->ExceptionCheck();
+    if (flag) {
+      jthrowable jerror = jenv->ExceptionOccurred();
```

Comment:
do you really need to retrieve the exception and set it again? when I tested if you don't clear, there is no need to Throw again.

Reply:
good point, since I'm re-throwing the same error I guess I might as well just return without clearing

https://www.developer.com/java/data/exception-handling-in-jni.html

Reply:
I'm seeing this error if I don't rethrow, which looks good

Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: -1
[Dynamic-linking native method java.lang.Throwable.getStackTraceDepth ... JNI]
[Dynamic-linking native method java.lang.Throwable.getStackTraceElement ... JNI]
	at com.microsoft.ml.lightgbm.lightgbmlibJNI.LGBM_DatasetCreateFromCSRSpark(Native Method)
	at com.microsoft.ml.lightgbm.lightgbmlib.LGBM_DatasetCreateFromCSRSpark(lightgbmlib.java:286)
	at program.generateSparseDataset(program.java:24)
	at program.main(program.java:59)